### ISSUE TITLE
[FR] ollama

### ISSUE THREAD

**Issue by annieappflowy** [AppFlowy-IO/AppFlowy#4157]
### Description

Could we add the feature of supporting ollama in Appflowy ? It has an API and could be a self-hosted and open source alternative to the OpenAI API 

### Impact

an API and could be a self-hosted and open source alternative to the OpenAI API

**Comment by Clovis-krz** [comment]
Well I would use it exactly like the OpenAI Writer feature. 
Instead of sending query to Chatgpt url it's send to your ollama API url. 
There could be an option in settings or while prompting to select which of your installed model to select.
Now ollama even supports multimodal models so we could even select in the editor a specific picture and ask to write a description about it (aka using llava).
There is so much more possible like having a chatbot on the side, ingest documents and write a summary...

**Comment by paulonasc** [comment]
👋 new to contributing here, can I pick this feature up? @annieappflowy

**Comment by annieappflowy** [comment]
@paulonasc , thanks for offering the help. Since it's very complex and critical, this task will be handled by the appflowy team. We encourage you to look for alternative issues that may interest you.

**Comment by bprolifcdev** [comment]
Is this still in the pipeline ? Should be really easy to implement right ?

**Comment by Snuupy** [comment]
I saw in the docker-compose.yml you can specify your own API URL, not sure if anyone's tried that yet

**Comment by oc013** [comment]
@Snuupy could you point me to where you see it? I'm not seeing it here 
https://github.com/AppFlowy-IO/AppFlowy/blob/main/frontend/scripts/docker-buildfiles/docker-compose.yml

**Comment by Snuupy** [comment]
https://github.com/AppFlowy-IO/AppFlowy-Cloud/blob/945f1379fa453a1e4860d1b475f134fdfc76efd4/docker-compose.yml#L108C8-L108C9 @oc013 

looks like they removed it?

???

**Comment by oc013** [comment]
Thank you @Snuupy I will need to try to understand what that repo for, I thought everything was here. (I just found AppFlowy yesterday looking for something that integrates with Ollama :smile: )

https://github.com/AppFlowy-IO/AppFlowy-Cloud/pull/512

**Comment by themantalope** [comment]
@Snuupy - have you tried this feature? are you running a self hosted cloud instance?

**Comment by Snuupy** [comment]
> @Snuupy - have you tried this feature? are you running a self hosted cloud instance?

no, not yet. I paused on continuing to test appflowy back when you could not view/edit databases on mobile but it looks like that was just recently merged. I also don't like the AI paywall so I'm hesitant to sink more time into this.

**Comment by annieappflowy** [comment]
@oc013 , We didn’t have the time to do it. If you’ve checked our recent release notes, there aren’t any AI-related features...
This is the current status from our side regarding this request:
<img width="713" alt="image" src="https://github.com/user-attachments/assets/d33979a0-0d35-487d-815e-e28702c5da78">

The next to-do for the self-hosting solution is to support self-hosting our web app. #5920

**Comment by khorshuheng** [comment]
I have added a reply in https://github.com/AppFlowy-IO/AppFlowy-Cloud/issues/943 which provides more information on this.

**Comment by annieappflowy** [comment]
Supported in the upcoming release (v0.8.7)
https://appflowy.com/guide/appflowy-local-ai-ollama
